x-pack/plugins/endpoint/public/applications/endpoint/view/managing/details.tsx

```diff
@@ -0,0 +1,159 @@
+
+const HostDetails = memo(() => {
+  const details = useManagementListSelector(detailsData);
+  if (details === undefined) {
+    return null;
+  }
+
+  const detailsResultsUpper = useMemo(() => {
+    return [
+      {
+        title: i18n.translate('xpack.endpoint.management.details.os', {
+          defaultMessage: 'OS',
+        }),
+        description: details.host.os.full,
+      },
+      {
+        title: i18n.translate('xpack.endpoint.management.details.lastSeen', {
+          defaultMessage: 'Last Seen',
+        }),
+        description: details['@timestamp'],
+      },
+      {
+        title: i18n.translate('xpack.endpoint.management.details.alerts', {
+          defaultMessage: 'Alerts',
+        }),
+        description: '0',
+      },
+    ];
+  }, [details]);
+
+  const detailsResultsLower = useMemo(() => {
+    return [
+      {
+        title: i18n.translate('xpack.endpoint.management.details.policy', {
+          defaultMessage: 'Policy',
+        }),
+        description: details.endpoint.policy.id,
+      },
+      {
+        title: i18n.translate('xpack.endpoint.management.details.policyStatus', {
+          defaultMessage: 'Policy Status',
+        }),
+        description: 'active',
+      },
+      {
+        title: i18n.translate('xpack.endpoint.management.details.ipAddress', {
+          defaultMessage: 'IP Address',
+        }),
+        description: details.host.ip,
+      },
+      {
+        title: i18n.translate('xpack.endpoint.management.details.hostname', {
+          defaultMessage: 'Hostname',
+        }),
+        description: details.host.hostname,
+      },
+      {
+        title: i18n.translate('xpack.endpoint.management.details.sensorVersion', {
+          defaultMessage: 'Sensor Version',
+        }),
+        description: details.agent.version,
+      },
+    ];
+  }, [details.agent.version, details.endpoint.policy.id, details.host.hostname, details.host.ip]);
+
+  return (
+    <>
+      <EuiDescriptionList
+        type="column"
+        listItems={detailsResultsUpper}
+        data-test-subj="managementDetailsUpperList"
+      />
+      <EuiHorizontalRule margin="s" />
+      <EuiDescriptionList
+        type="column"
+        listItems={detailsResultsLower}
+        data-test-subj="managementDetailsLowerList"
+      />
+    </>
+  );
+});
+
+export const ManagementDetails = () => {
+  const history = useHistory();
+  const { notifications } = useKibana();
+  const queryParams = useManagementListSelector(uiQueryParams);
+  const { selected_host: selectedHost, ...queryParamsWithoutSelectedHost } = queryParams;
+  const details = useManagementListSelector(detailsData);
+  const error = useManagementListSelector(detailsError);
+
+  const handleFlyoutClose = useCallback(() => {
+    history.push(urlFromQueryParams(queryParamsWithoutSelectedHost));
+  }, [history, queryParamsWithoutSelectedHost]);
+
+  useEffect(() => {
```

Comment:
@oatkiller @paul-tavares @parkiino Let's discuss if it makes sense to follow this pattern in future PRs. It's a little odd to use redux/middleware and also introduce side-effects to the functional components. At first, I thought we could run this side-effect in the middleware by doing something like:
```javascript
    try {
      const response = await coreStart.http.get(`/api/endpoint/metadata/${selectedHost}`);
    } catch (e) {
      notifications.toasts.addDanger({
        title: i18n.translate('key', {
          defaultMessage: 'Error',
        }),
      });
    }
```
But the problem with this approach is that we won't be able to create more complex toasts with html elements.

The other option is to use the [EUIToast ](https://elastic.github.io/eui/#/display/toast) component and use redux state to determine when to show it and hide it.

Reply:
seems like the react version (EUIToast) would be easier to test for

Reply:
I guess I'm ok with building the toast ourselves and displaying it that way. I thought that using the kibana service would be best in order to plug into a more global management of toasts (have not looked, but I assume it protects from not having an avalanche of toasts at any one time).

Although `useEffect` was used above, we did not have to use it - we could have called (I think) the `notifications.toasts.danger` directly and handle the logic around whether the error was already shown or not ourselves - `useEffect` just made it easier to implement since it allows for checking dependencies

**Re: Handling of error conditions**
My my mind, this belongs in the view since it is UI feedback to the user. Pushing it out to another module that is detached from the View and possibly unaware of what else needs to happen would not be good IMO.

Reply:
@paul-tavares I agree with you RE: error handling. I think we just need to decide whether to use the service or add our own Toasts component using `EuiGlobalToastList`, which the notifications service uses under the hood. I guess the advantage of rolling our own is that we could store the toasts in the redux store and have more control. It may facilitate testing too, like Robert suggested. I don't have a strong preference. I just wanted to have the discussion.

Reply:
@peluja1012 I'm good if we want to roll our own process/management of toasts. Personally, I don't see the value in it since the platform already provides a service for it, but am open with going with other alternatives.

Also, keep in mind that normally toasts will be visible even after a component is unmounted (they are kind of "global"). So if we roll our own solution we will need to keep that in mind because I am not sure what the `EuiGlobalToastList` does if you mount that component in our view (ex. flyout) and then close out (`unmount`) the flyout soon after - which could be the case in create flows where we might use a Flyout and close it after creating the item. The same case would apply if user navigates to another Kibana app - would the toast abruptly be closed? Just some behaviors to think about as we decide how to proceed. 

(FYI: did a quick search on `x-pack` and there is not much usage of `EuiGlobalToastList`)

Reply:
@peluja1012 @paul-tavares @oatkiller 

I created this discussion ticket where we can prioritize the importance of going another route with toasts within our application.  https://github.com/elastic/endpoint-app-team/issues/254

Feel free to add anything to the ticket.